### ISSUE TITLE
Wrong error text for definite-assignment-assertion on ambient variable declaration

### ISSUE THREAD

**Issue by ajafff** [microsoft/TypeScript#41965]
**TypeScript Version:**  4.2.0-dev.20201211

**Search Terms:** definite assignment

**Expected behavior:**

`A definite assignment assertion '!' is not permitted in this context.(1255)` on both assertions

**Actual behavior:**

`Declarations with definite assignment assertions must also have type annotations.(1264)` on both assertions, although both have a type annotation.

**Related Issues:**

**Code**
```ts
declare namespace foo {
	var v!: number;
}
declare var v!: number;
```

**Playground Link:** [Provided](https://www.typescriptlang.org/play?ts=4.2.0-dev.20201211#code/CYUwxgNghgTiAEA7KBbEBnADlMCBmA9gfAN4BQAkAG6zxUCEAXEgK4oBGIMA3GQL5lQkWAhow6TVhy7cgA)
